servicetalk-gradle-plugin-internal/src/main/groovy/io/servicetalk/gradle/plugin/internal/ServiceTalkLibraryPlugin.groovy

```diff
@@ -74,9 +76,58 @@ final class ServiceTalkLibraryPlugin extends ServiceTalkCorePlugin {
             from components.java
             artifact(javadocJar)
             artifact(sourcesJar)
+            versionMapping {
+              usage('java-api') {
+                fromResolutionOf('runtimeClasspath')
+              }
+              usage('java-runtime') {
+                fromResolutionResult()
+              }
+            }
+            pom {
+              name = project.name
+              description = 'A networking framework that evolves with your application'
+              url = 'https://servicetalk.io'
+              licenses {
+                license {
+                  name = 'The Apache License, Version 2.0'
+                  url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                }
+              }
+              developers {
+                developer {
+                  id = 'servicetalk-project-authors'
+                  name = 'ServiceTalk project authors'
+                  email = 'servicetalk-oss@group.apple.com'
+                }
+              }
+              scm {
+                connection = 'scm:git:git://github.com/apple/servicetalk.git'
+                developerConnection = 'scm:git:ssh://github.com:apple/servicetalk.git'
+                url = 'https://github.com/apple/servicetalk'
+              }
+            }
+          }
+        }
+        repositories {
+          maven {
+            def releasesRepoUrl = "https://oss.sonatype.org/service/local/staging/deploy/maven2"
+            def snapshotsRepoUrl = "https://oss.sonatype.org/content/repositories/snapshots"
+            url = version.endsWith('SNAPSHOT') ? snapshotsRepoUrl : releasesRepoUrl
+            credentials {
+              username = System.getenv("SONATYPE_USER")
+              password = System.getenv("SONATYPE_TOKEN")
+            }
           }
         }
       }
+      signing {
+        required findProperty("signingKey") != null
+        def signingKey = findProperty("signingKey")
+        def signingPassword = findProperty("signingPassword")
+        useInMemoryPgpKeys(signingKey, signingPassword)
+        sign publishing.publications.mavenJava
+      }
```

Comment:
Add the following section right after `signing` block:

```groovy
tasks.withType(AbstractPublishToMaven) {
  onlyIf {
    // Disable all tasks that try to publish something else, expect defined "mavenJava" publication.
    // That could be automatically configured "pluginMaven" publication for gradle plugins that are required
    // only for Gradle Plugin Portal and should not be published to Maven Central
    publication == publishing.publications.mavenJava
  }
}
```
**Note**: you need to import `AbstractPublishToMaven`, otherwise it doesn't work.

It helps to remove the error we saw:
```
Multiple publications with coordinates 'io.servicetalk:servicetalk-gradle-plugin-internal:0.20.0-SNAPSHOT' are published to repository 'mavenLocal'. The publications will overwrite each other!
```
If we do not do this, `pluginMaven` publication overrides the jar, pom, and maven-metadata files generated by  `mavenJava` publication. It removes `source` and `javadoc` metadata from `maven-metadata.xml` 😞

Reply:
Just verified, signatures are intact and plugin meta-data exists in the JAR, so this approach does seem to work indeed! 🥇